serve_test.go

```diff
@@ -188,6 +189,46 @@ func TestServer_Run(t *testing.T) {
 	})
 }
 
+func TestServer_RunTLS(t *testing.T) {
+	// This is not a standard test, it is here to ensure that the server can run.
+	// Please do not run this kind of test for your controllers, it is NOT unit testing.
+	tt := []struct {
+		name string
+		opts []func(*Server)
+	}{
+		{
+			name: "can run TLS server with TLS config",
+			opts: []func(*Server){WithTLSConfig(&tls.Config{}), WithoutLogger()},
```

Comment:
This fails to start up the server. 

If you remove the go func and check for the err you can see it fail with. 

```
open : no such file or directory
```

As in this case no file is given: `TLSConfig.Certificates` or `TLSConfig.GetCertificate` has to be populated

Reply:
Thanks, I'll try to improve the tests.